### DISCUSSION TITLE
Setting environment for babel.config.js in react native

### DISCUSSION THREAD

**Discussion by swizes** [babel/babel#14409] [Q&A]
I want to achieve basically the below architecture for babel.config.js for a react-native app

module.exports = {
  presets: ['module:metro-react-native-babel-preset'],
  env: {
    production: {
      plugins: [someProdPlugins],
    },
 development: {
      plugins: [someDevPlugins],
    },
 staging: {
      plugins: [someStagingPlugins],
    },
  },
};

How can i set the build environment so the babel uses the right plugins

**api.env() always returns development**

Thanks in advance

**Comment by nicolo-ribaudo** [comment]
You can set the `BABEL_ENV=production` environment variable when calling Babel.
How to do that depends on your OS and shell.
When using most Linux shells, you can do it with `BABEL_ENV=production the-command-you-use-to-compile-your-app`
